core/chains/evm/types/models.go

```diff
@@ -153,6 +155,18 @@ func (h *Head) ChainHashes() []common.Hash {
 	return hashes
 }
 
+func (h *Head) ChainId() *big.Int {
+	return h.EVMChainID.ToInt()
+}
+
+func (h *Head) IsChainIdNil() bool {
```

Comment:
The reason i created these functions was because, checking for nil values or doing a comparison with generics is tedious and would result in messier code.

Reply:
I think that this approach only hides the problem. It is possible to have a generic type which is not nilable: how is this method expected to be implemented in those cases?

Reply:
@jmank88 That is a good point. In the context of head, i would suppose that if the Head is not nilable, It would return true in those cases

For the context: 
https://github.com/smartcontractkit/chainlink/pull/9311/files#diff-3ad9f632ee4d20c3624ac906af5da884edf45cca8de21b0e8f98b7c515495d5fL128

Maybe there is a better way of checking for nullity or for the same chain of the head and client head?

Reply:
I'm not following. Why would a non-nilable type ever return true from an `IsNil` method?
I think that having `nil` in the name is presuming too much from the start. Could something like `HasChainID` serve the same purpose?
```suggestion
func (h *Head) HasChainID() bool {
```

Reply:
Yup thanks for the improvement of naming. `HasChainId` is a much better naming!